x/stake/simulation/msgs.go

```diff
@@ -0,0 +1,253 @@
+package simulation
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/mock"
+	"github.com/cosmos/cosmos-sdk/x/mock/simulation"
+	"github.com/cosmos/cosmos-sdk/x/stake"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/crypto"
+)
+
+// SimulateMsgCreateValidator
+func SimulateMsgCreateValidator(m auth.AccountMapper, k stake.Keeper) simulation.TestAndRunTx {
+	return func(t *testing.T, r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, keys []crypto.PrivKey, log string, event func(string)) (action string, err sdk.Error) {
+		denom := k.GetParams(ctx).BondDenom
+		description := stake.Description{
+			Moniker: simulation.RandStringOfLength(r, 10),
+		}
+		key := keys[r.Intn(len(keys))]
```

Comment:
Maybe I'm bonkers, but I think due to the nature of this type of compact statement, our code would be more clear if we expanded this to a function to separate out concerns a bit.  

```golang
key := RandomKey(r, keys) 

func RandomKey(r *rand.Rand, keys []crypto.PrivKey) key {
    return keys[r.Intn(
        len(keys)
    )]
}
```

I've always found it a bit irritating reading tests with the rand function because of lines like this, to separate out would make my mind and heart at ease

Reply:
(obv. applies to all statements like this in this PR)

Reply:
Agreed, that's cleaner. Updated.